doc/api/modules.md

```diff
@@ -154,14 +154,16 @@ require(X) from module at path Y
 1. If X is a core module,
    a. return the core module
    b. STOP
-2. If X begins with '/'
+2. If split(X, '/')[0] is a core module,
```

Comment:
Perhaps rather - `If the first path segment of X is a core module`.

Reply:
does that imply backslash would also work? I wouldn't want people to get confused and implement this with backslashes being allowed.

Reply:
Perhaps the implementation should check backslashes in Windows then?

A clarification then might be - `If the first "/" or "\" separated path segment of X is a core module, ...`.

There are a number of places we went back and forth on backslash checking in the ESM implementation, but stuck with `/` checking as the primary mechanism for package name and pathing validations, as brought up by @jkrems as well.

The URL spec usually just treats all backslashes as forward slashes as a general rule preprocessing I believe... something similar might work as a blanket fix in our code paths too.

Reply:
We should probably stick with `If the first '/' separated path segment of X...` for now, and make handling backslashes something we do over all resolver paths if we ever want to explicitly handle them.

Reply:
Or alternatively, if we do decide to blanket handle backslashes, that is a decision that can be rolled out gradually, but should still be made generally...